src/Thentos/Transaction/Transactions.hs

```diff
@@ -49,6 +49,10 @@ assertUser mUid user = polyQuery $ ask >>= \ db ->
        | userFacetExists (^. userEmail) mUid user db -> throwT UserEmailAlreadyExists
        | True -> return ()
 
+-- | Assert that no user with the same name or email address already exists in the DB.
+trans_assertUserIsNew :: (AsDB db) => User -> ThentosQuery db ()
+trans_assertUserIsNew = assertUser Nothing
```

Comment:
see earlier comment on the missing exception in `addUnconfirmedUserWithId`.  if you fix that, this here may become unnecessary.  in that case i'd like to keep the api simple.  perhaps `lookupUser` can cover for this one?